cirq/testing/random_circuit.py

```diff
@@ -52,40 +59,48 @@ def random_circuit(qubits: Union[Sequence[ops.QubitId], int],
         op_density: the expected proportion of qubits that are acted on in any
             moment.
         gate_domain: The set of gates to choose from, with a specified arity.
-
     Raises:
         ValueError:
             * op_density is not in (0, 1).
             * gate_domain is empty.
             * qubits is an int less than 1 or an empty sequence.
-
     Returns:
         The randomly generated Circuit.
-    """
-    if not 0 < op_density < 1:
-        raise ValueError('op_density must be in (0, 1).')
-    if gate_domain is None:
-        gate_domain = DEFAULT_GATE_DOMAIN
-    if not gate_domain:
-        raise ValueError('gate_domain must be non-empty')
-    max_arity = max(gate_domain.values())
-
-    if isinstance(qubits, int):
-        qubits = tuple(ops.NamedQubit(str(i)) for i in range(qubits))
-    n_qubits = len(qubits)
-    if n_qubits < 1:
-        raise ValueError('At least one qubit must be specified.')
-
-    moments = [] # type: List[ops.Moment]
-    for _ in range(n_moments):
-        operations = []
-        free_qubits = set(q for q in qubits)
-        while len(free_qubits) >= max_arity:
-            gate, arity = choice(tuple(gate_domain.items()))
-            op_qubits = sample(free_qubits, arity)
-            free_qubits.difference_update(op_qubits)
-            if random() <= op_density:
-                operations.append(gate(*op_qubits))
-        moments.append(ops.Moment(operations))
-
-    return Circuit(moments)
+        """
+
+        if not 0 < self.op_density < 1:
+            raise ValueError('op_density must be in (0, 1).')
+        if self.gate_domain is None:
+            self.gate_domain = DEFAULT_GATE_DOMAIN
+        if not self.gate_domain:
+            raise ValueError('gate_domain must be non-empty')
+        max_arity = max(self.gate_domain.values())
+
+        if isinstance(self.qubits, int):
+            self.qubits = tuple(
+                ops.NamedQubit(str(i)) for i in range(self.qubits))
+        n_qubits = len(self.qubits)
+        if n_qubits < 1:
+            raise ValueError('At least one qubit must be specified.')
+
+        moments = [] # type: List[ops.Moment]
+        for _ in range(self.n_moments):
+            operations = []
+            free_qubits = set(q for q in self.qubits)
+            while len(free_qubits) >= max_arity:
+                gate, arity = choice(tuple(self.gate_domain.items()))
+                op_qubits = sample(free_qubits, arity)
+                free_qubits.difference_update(op_qubits)
+                if random() <= self.op_density:
+                    operations.append(gate(*op_qubits))
+            moments.append(ops.Moment(operations))
+
+        return Circuit(moments)
+
+    def random_superposition(self):
```

Comment:
This method does not have a clear name. It's generating a superposition using the circuit as a source of entropy? But the randomness of the superposition will strongly depend on the gate set of the circuit.

I don't think this method is needed. What purpose did you have in mind for it?

Reply:
After talking to Kevin, i will be getting rid of this class and following his (simpler) suggestion for `random_superposition`. I was going to get rid of this on the most recent commit.